segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableAttributes.java

```diff
@@ -40,6 +40,17 @@
      */
     public static final UUID TOTAL_ENTRY_COUNT = new UUID(CORE_ATTRIBUTE_ID_PREFIX, TABLE_ATTRIBUTES_START_OFFSET + 3);
 
+    /**
+     * Defines an attribute that is used to store the offset of a (Table) Segment where compaction has last run at.
+     */
+    public static final UUID COMPACTION_OFFSET = new UUID(CORE_ATTRIBUTE_ID_PREFIX, TABLE_ATTRIBUTES_START_OFFSET + 4);
```

Comment:
Just because you compact (copy entries), you can't truncate right away. The Index still points to the original entry locations and if we truncate we will not be able to serve that data. A subsequent iteration of the processor will index the copied entries and it will repoint the index to the new locations. Only then it is safe to truncate this data.